### PR TITLE
Refactor: Remove redundant logger from StorageClusterReconciler methods

### PR DESCRIPTION
**Pull request by iamniting**
refactor StorageClusterReconciler methods to use
StorageClusterReconciler logger and remove
logr.Logger type argument from methods.

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

**Review by umangachapagain — COMMENTED** [review]
`Reconcile()` method in `reconcile.go` does
```
reqLogger := r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
```
which adds more context to logs. Using `r.Log` seems to ignore that extra context.
So we can remove that line and add context in-line while logging or find a way to globally initialize `r.Log` with more context.

WDYT?

**Comment by iamniting** [comment]
> `Reconcile()` method in `reconcile.go` does
> 
> ```
> reqLogger := r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
> ```
> 
> which adds more context to logs. Using `r.Log` seems to ignore that extra context.
> So we can remove that line and add context in-line while logging or find a way to globally initialize `r.Log` with more context.
> 
> WDYT?

I agree we should do that

**Comment by umangachapagain** [comment]
@iamniting Can you also refactor OCSInitialization and PV Reconciler? Seems like a small change and would make the usage of `r.Log` uniform across codebase. Thanks.

**Comment by iamniting** [comment]
> @iamniting Can you also refactor OCSInitialization and PV Reconciler? Seems like a small change and would make the usage of `r.Log` uniform across codebase. Thanks.

@umangachapagain Done

File: controllers/storagecluster/reconcile.go [hunk 1]
```diff
@@ -112,13 +112,13 @@ var validTopologyLabelKeys = []string{
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *StorageClusterReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	reqLogger := r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	r.Log = r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
```
Comment (raghavendra-talur) [inline controllers/storagecluster/reconcile.go:116]: Please hold onto the old logger and restore it at the end of the function using a defer func. We don't want to be logging with the stale namespace and name outside of the reconcile function.

File: controllers/ocsinitialization/ocsinitialization_controller.go [hunk 2]
```diff
@@ -190,14 +190,14 @@ func (r *OCSInitializationReconciler) ensureToolsDeployment(initialData *ocsv1.O
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *OCSInitializationReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	reqLogger := r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
-	reqLogger.Info("Reconciling OCSInitialization")
+	r.Log = r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	r.Log.Info("Reconciling OCSInitialization")
```
Comment (raghavendra-talur) [inline controllers/ocsinitialization/ocsinitialization_controller.go:194]: Similar change here. Backup the old logger.

File: controllers/persistentvolume/reconcile.go [hunk 3]
```diff
@@ -13,38 +12,38 @@ import (
...
-type ensureFunc func(*corev1.PersistentVolume, logr.Logger) error
+type ensureFunc func(*corev1.PersistentVolume) error
 
 // +kubebuilder:rbac:groups=core,resources=persistentvolumes,verbs=get;list;watch;update;patch
 // +kubebuilder:rbac:groups=storage.k8s.io,resources=storageclasses,verbs=get
 
 // Reconcile ...
 func (r *PersistentVolumeReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	reqLogger := r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	r.Log = r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
```
Comment (raghavendra-talur) [inline controllers/persistentvolume/reconcile.go:22]: Similar change here.

**Review by raghavendra-talur — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/reconcile.go (hunk 1 above)
Comment (iamniting) [reply controllers/storagecluster/reconcile.go:119]: Done

File: controllers/ocsinitialization/ocsinitialization_controller.go (hunk 2 above)
Comment (iamniting) [reply controllers/ocsinitialization/ocsinitialization_controller.go:198]: Done

File: controllers/persistentvolume/reconcile.go (hunk 3 above)
Comment (iamniting) [reply controllers/persistentvolume/reconcile.go:25]: Done

File: controllers/ocsinitialization/ocsinitialization_controller.go (hunk 2 above)
Comment (raghavendra-talur) [reply controllers/ocsinitialization/ocsinitialization_controller.go:198]: This looks good to me.

File: controllers/persistentvolume/reconcile.go (hunk 3 above)
Comment (raghavendra-talur) [reply controllers/persistentvolume/reconcile.go:25]: This looks good to me.

**Review by raghavendra-talur — APPROVED** [review]

**Review by jarrpa — APPROVED** [review]
/lgtm

**Pull request merged**